### PR TITLE
Chroma memory

### PR DESCRIPTION
**Pull request by jhouseke**
### Background
Project needs more open source vector embedding db options.

### Changes
Implemented Chroma database.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [X] I have thouroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: scripts/config.py
Comment (nponeccop) [inline scripts/config.py]: Add a CRLF at the end of file

File: scripts/memory/chroma.py
Comment (nponeccop) [inline scripts/memory/chroma.py]: Here too

**Review by nponeccop — CHANGES_REQUESTED** [review]

File: scripts/config.py
Comment (jhouseke) [reply scripts/config.py]: should be fixed now!

File: scripts/memory/chroma.py
Comment (jhouseke) [reply scripts/memory/chroma.py]: fixed this one too

**Review by nponeccop — DISMISSED** [review]

File: scripts/config.py
Comment (aslafy-z) [inline scripts/config.py]: Please use underscores instead of dashes for environment variables

Comment (jhouseke) [reply scripts/config.py]: done and done

**Comment by jhouseke** [comment]
Resolved conflicts, thank you for heads up

**Review by nponeccop — APPROVED** [review]

**Review by nponeccop — DISMISSED** [review]

**Comment by nponeccop** [comment]
@jhouseke There are conflicts now

**Comment by cryptidv** [comment]
@jhouseke Any updates on this?

**Comment by jhouseke** [comment]
resolving now

**Comment by jhouseke** [comment]
PR no longer has conflicts. Anything else I need to be aware of?

**Comment by nponeccop** [comment]
@jhouseke CI is red now.

**Comment by jhouseke** [comment]
removed whitespace to resolve W293

**Comment by HammadB** [comment]
Any progress on this - would love to use chroma with autogpt?

**Comment by jhouseke** [comment]
@nponeccop any chance this can get merged in? thanks!

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by ntindle** [comment]
Hey, I've marked this as don't merge until the Memory Fixes are in. Sorry to keep it on hold longer, just no real way to test/validate functionality until the fixes are in

**Comment by Pwuts** [comment]
There is currently no prospect of this getting merged, so unfortunately I'll have to close it. The structure of the codebase and the application has changed a lot, so merging it would be a lot of work, and before we start adding vector stores, we need a good high-level design for memory *use*. This will be tracked in #3536, which needs updating as I write this.

There's a good chance Chroma will eventually be added, and we really do appreciate the effort that went into this PR. If you're still interested in (contributing to) the project, keep an eye on the [#dev-autogpt](https://discord.com/channels/1092243196446249134/1095817829405704305) channel. We've been making good progress in the past weeks, and my current #1 todo is making sure contributors can meaningfully contribute again.

**Pull request closed without merging**
